pkg/plugin/aggregation/run.go

```diff
@@ -150,18 +150,46 @@ func Run(client kubernetes.Interface, plugins []plugin.Interface, cfg plugin.Agg
 	}()
 
 	// 4. Launch each plugin, to dispatch workers which submit the results back
-	for _, p := range plugins {
-		cert, err := auth.ClientKeyPair(p.GetName())
-		if err != nil {
-			return errors.Wrapf(err, "couldn't make certificate for plugin %v", p.GetName())
-		}
-		logrus.WithField("plugin", p.GetName()).Info("Running plugin")
-		if err = p.Run(client, cfg.AdvertiseAddress, cert); err != nil {
-			return errors.Wrapf(err, "error running plugin %v", p.GetName())
+	go func() {
+		pluginsRun := map[int]bool{}
+		for {
```

Comment:
Hi @johnSchnake,

You may want to consider using a Go context and channels to select on a channel buffered to the size of the plug-ins and `ctx.Done` in case the operation is cancelled.

Reply:
Hi @akutz , that is an interesting idea but would that actually be necessary here since all this is running in containers in the cluster? A "cancel" in that context would mean the user is stopping the run and we are just blowing up the container entirely, right? Or is there another way you can see that being used?

Reply:
Hi @johnSchnake,

I didn't have any clear idea on how this was being used. When I see the `for {` pattern I usually ask myself if it should be cancellable, and if so, does it make sense to use a Go context. You know the call stack better than I do, so please do whatever you think makes sense.